File: Ejercicios/4. WeatherData/Net/Before/WeatherData/ForecastDisplay.cs
using System;
using System.Text;

namespace WeatherData
{
    public class ForecastDisplay : IDisplayElement
    {
        private float currentPressure = 29.92f;

        private float lastPressure;

        public void Update(float temperature, float humidity, float pressure)
        {
            this.lastPressure = this.currentPressure;
            this.currentPressure = pressure;
            Display();
        }

        public void Display()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("Forecast: ");

            if (this.currentPressure > this.lastPressure)
            {
                sb.Append("Improving weather on the way!");
            }
            else if (this.currentPressure == this.lastPressure)
            {
                sb.Append("More of the same");
            }
            else if (this.currentPressure < this.lastPressure)
            {
                sb.Append("Watch out for cooler, rainy weather");
            }
            Console.WriteLine(sb.ToString());
        }
    }
}

File: Ejercicios/4. WeatherData/Java/Before/src/main/weather/ForecastDisplay.java
package weather;

public class ForecastDisplay implements DisplayElement {
    private float currentPressure = 29.92f;

    private float lastPressure;

    public void update(float temperature, float humidity, float pressure)
    {
        this.lastPressure = this.currentPressure;
        this.currentPressure = pressure;
        display();
    }

    public void display()
    {
        StringBuilder sb = new StringBuilder();

        sb.append("Forecast: ");

        if (this.currentPressure > this.lastPressure)
        {
            sb.append("Improving weather on the way!");
        }
        else if (this.currentPressure == this.lastPressure)
        {
            sb.append("More of the same");
        }
        else if (this.currentPressure < this.lastPressure)
        {
            sb.append("Watch out for cooler, rainy weather");
        }
        System.out.println(sb.toString());
    }
}
